libs/core/type_support/tests/unit/is_trivially_relocatable.cpp

```diff
@@ -125,6 +125,41 @@ static_assert(
 static_assert(
     hpx::is_trivially_relocatable_v<explicitly_trivially_relocatable_2>);
 
+// c-v-ref-array qualified versions of explicitly declared trivially relocatable
+// types are trivially relocatable
+
+static_assert(
+    hpx::is_trivially_relocatable_v<explicitly_trivially_relocatable_1 const>);
+static_assert(hpx::is_trivially_relocatable_v<
+    explicitly_trivially_relocatable_1 volatile>);
+static_assert(hpx::is_trivially_relocatable_v<
+    explicitly_trivially_relocatable_1 const volatile>);
+static_assert(
+    hpx::is_trivially_relocatable_v<explicitly_trivially_relocatable_1&>);
+static_assert(
+    hpx::is_trivially_relocatable_v<explicitly_trivially_relocatable_1&&>);
+static_assert(
+    hpx::is_trivially_relocatable_v<explicitly_trivially_relocatable_1[]>);
+static_assert(
+    hpx::is_trivially_relocatable_v<explicitly_trivially_relocatable_1[10]>);
+
+// Chain of c-v-ref-array qualifiers are supported
+static_assert(hpx::is_trivially_relocatable_v<
+    explicitly_trivially_relocatable_1[10][10]>);
+static_assert(hpx::is_trivially_relocatable_v<
+    explicitly_trivially_relocatable_1 const[10]>);
+static_assert(hpx::is_trivially_relocatable_v<
+    explicitly_trivially_relocatable_1 volatile[10]>);
+static_assert(hpx::is_trivially_relocatable_v<
+    explicitly_trivially_relocatable_1 const volatile[10]>);
+static_assert(hpx::is_trivially_relocatable_v<
+    explicitly_trivially_relocatable_1 (&)[10]>);
+static_assert(
+    hpx::is_trivially_relocatable_v<explicitly_trivially_relocatable_1(
+        &&)[10]>);
+static_assert(hpx::is_trivially_relocatable_v<
+    explicitly_trivially_relocatable_1 const volatile&>);
+
```

Comment:
Above, I suggest changing the answers for reference types.
I also suggest checking the answers for `void`, `int()`, `int(&)()`, which should all be `false`, to match `std::is_trivially_copyable`.